### PR TITLE
1.0 fix various bugs and tests

### PR DESCRIPTION
**Pull request by jranson**

### CODE REVIEW

File: internal/config/loader_test.go [hunk 1]
```diff
@@ -39,3 +39,475 @@ func TestLoadConfiguration(t *testing.T) {
...
+		t.Errorf("expected 57821, got %d", ProxyServer.ListenPort)
+	}
+
+	if ProxyServer.ListenAddress != "test" {
+		t.Errorf("expected test, got %s", ProxyServer.ListenAddress)
+	}
+
+	// Test Metrics Server
+	if Metrics.ListenPort != 57822 {
+		t.Errorf("expected 57821, got %d", Metrics.ListenPort)
```
Comment (LimitlessEarth) [inline internal/config/loader_test.go]: 57822?

File: internal/config/loader_test.go [hunk 2]
```diff
@@ -39,3 +39,475 @@ func TestLoadConfiguration(t *testing.T) {
...
+	}
+
+	if Metrics.ListenAddress != "metrics_test" {
+		t.Errorf("expected test, got %s", Metrics.ListenAddress)
```
Comment (LimitlessEarth) [inline internal/config/loader_test.go]: test -> metrics_test?

File: internal/config/loader_test.go (hunk 1 above)
Comment (LimitlessEarth) [reply internal/config/loader_test.go]: Oh you set constants in the next Commit. NVM.

File: internal/config/loader_test.go (hunk 2 above)
Comment (LimitlessEarth) [reply internal/config/loader_test.go]: Oh you set constants in the next Commit. NVM.

File: internal/config/loader_test.go (hunk 1 above)
Comment (jranson) [reply internal/config/loader_test.go]: Actually it's not the constants that makes this work, but rather the configuration files in `testdata/test.full.conf` and `testdata/test.empty.conf`. This test loads those files and expects them to resolve specific values.

**Pull request merged**